bolt/src/Relocation.cpp

```diff
@@ -132,6 +132,71 @@ size_t getSizeForTypeAArch64(uint64_t Type) {
   }
 }
 
+bool skipRelocationProcessX86(uint64_t Type, uint64_t Contents) {
+  return false;
+}
+
+bool skipRelocationProcessAArch64(uint64_t Type, uint64_t Contents) {
+  auto IsMov = [](uint64_t Contents) -> bool {
+    // The bits 28-23 are 0b100101
+    if ((Contents & 0x1f800000) == 0x12800000)
+      return true;
+    return false;
+  };
+
+  auto IsB = [](uint64_t Contents) -> bool {
+    // The bits 31-26 are 0b000101
+    if ((Contents & 0xfc000000) == 0x14000000)
+      return true;
+    return false;
+  };
+
+  auto IsNop = [](uint64_t Contents) -> bool {
+    return Contents == 0xd503201f;
+  };
+
+  // The linker might eliminate the instruction and replace it with NOP, ignore
+  if (IsNop(Contents))
+    return true;
+
+  // The linker might perform TLS relocations relaxations, such as
+  // changed TLS access model (e.g. changed global dynamic model
+  // to initial exec), thus changing the instructions. The static
+  // relocations might be invalid at this point and we might no
+  // need to proccess these relocations anymore.
+  // More information could be find by searching
+  // elfNN_aarch64_tls_relax in bfd
+  switch (Type) {
+  default:
+    break;
+  case ELF::R_AARCH64_TLSIE_LD64_GOTTPREL_LO12_NC:
+  case ELF::R_AARCH64_TLSIE_ADR_GOTTPREL_PAGE21: {
+    if (IsMov(Contents))
+      return true;
+  }
+  }
+
+  // The ld might replace load/store instruction with jump and
+  // veneer due to errata 843419
+  // https://documentation-service.arm.com/static/5fa29fddb209f547eebd361d
+  // Thus load/store relocations for these instructions must be ignored
+  // NOTE: We only process GOT and TLS relocations this way since the
+  // addend used in load/store instructions won't change after bolt
+  // (it is important since the instruction in veneer won't have relocation)
+  switch (Type) {
+  default:
+    break;
+  case ELF::R_AARCH64_LD64_GOT_LO12_NC:
+  case ELF::R_AARCH64_TLSIE_LD64_GOTTPREL_LO12_NC:
+  case ELF::R_AARCH64_TLSDESC_LD64_LO12: {
+    if (IsB(Contents))
+      return true;
+  }
+  }
```

Comment:
Can we merge this in the same switch-case statement?

Reply:
Sure I may merge it, but we may have different cases with the same relocation like with R_AARCH64_TLSIE_LD64_GOTTPREL_LO12_NC . Plus it is more readable with the separated comments IMHO. Maybe we can stay with this variant? 
Also could you please answer about the other idea in the comments please?

Reply:
Oh I see what you mean. There is a path where R_AARCH64_TLSIE_LD64_GOTTPREL_LO12_NC fails the isMov check, and then it is checked again in the next switch-case statement. Sounds good.